### ISSUE TITLE
Strapi Upload AWS causing heroku build to fail

### ISSUE THREAD

**Issue by sgrund14** [strapi/strapi#2093]
**Informations**
- **Node.js version**: 10.11.0
- **npm version**: 6.4.1
- **Strapi version**:  3.0.0-alpha.14.2
- **Database**: MongoDB
- **Operating system**: Mac OSX Sierra

**What is the current behavior?**
When I install the `strapi-upload-aws` plugin, the heroku deployment of my app fails in the `build dependency` step, citing this error: ```Building dependencies
remote:        Installing node modules (package.json + package-lock)
remote:        npm ERR! path /tmp/build_bc8b453f104086bb93d57a0470f4da79/node_modules/.staging/strapi-ff077412/node_modules/@sailshq/lodash
remote:        npm ERR! code ENOENT
remote:        npm ERR! errno -2
remote:        npm ERR! syscall rename
remote:        npm ERR! enoent ENOENT: no such file or directory, rename '/tmp/build_bc8b453f104086bb93d57a0470f4da79/node_modules/.staging/strapi-ff077412/node_modules/@sailshq/lodash' -> '/tmp/build_bc8b453f104086bb93d57a0470f4da79/node_modules/.staging/@sailshq/lodash-805df19c'
remote:        npm ERR! enoent This is related to npm not being able to find a file.
remote:        npm ERR! enoent ```
The local build in `strapi start` works as expected.

**Steps to reproduce the problem**
1) `strapi new project && cd project`
2) `git init && git add heroku remote`
3) `git add -A && git commit -m "init commit" && `git push heroku master`. See that the build succeeds.
4) `npm install strapi-upload-aws-s3@alpha` && `git add, commit, push to heroku`. Build fails :(

**What is the expected behavior?**
The build should succeed in production

**Suggested solutions**
Not positive. Looks like there's something wrong with the build path in the strapi-aws plugin?

**Comment by lauriejim** [comment]
You don't use `--save` during your `npm install strapi-upload-aws-s3@alpha`

**Comment by lauriejim** [comment]
If you don't add it in you package.json, when the app will be deployed to Heroku, the dependence will not be installed. And if configure your app to use this provider in production env it will break your app cause the deps is missing.

**Comment by sgrund14** [comment]
@lauriejim Right. But the problem is that the production build is failing when I add it as a dependency in my package.json. So I can't even deploy it : (

**Comment by sgrund14** [comment]
@lauriejim I'm also actually noticing that I get the same error for any dependency I try to add. If I add anything to the package.json dependencies (`strapi-upload-aws-s3`, `strapi-upload-cloudinary`, even a `lodash.partition`) I get the error mentioned above.

**Comment by derrickmehaffy** [comment]
:stuck_out_tongue_winking_eye: @sgrund14

**Comment by rexfng** [comment]
@derrickmehaffy This does it for me. For anyone wanting to deploy to heroku I have a few tips.
1) get the heroku mongodb addon first before you deploy with the --as database flag ```heroku addons:create mongolab:sandbox --as DATABASE```
2) Install all the dependancy first such as sendgrid and s3 modules if you wanna use them. The deploy button in [this guide](https://medium.com/strapi/deploying-a-strapi-api-on-heroku-9c8b7809675c) doesn't have these plugin.
3) and yes remove the package-lock.json and git ignoring it
